subsys/bluetooth/mesh/friend.c

```diff
@@ -142,7 +148,7 @@ static void friend_clear(struct bt_mesh_friend *frnd)
 	if (frnd->last) {
 		/* Cancel the sending if necessary */
 		if (frnd->pending_buf) {
-			BT_MESH_ADV(frnd->last)->busy = 0U;
+			FRIEND_ADV(frnd->last)->adv.busy = 0U;
```

Comment:
I don't think this change is needed since `struct b t_mesh_adv adv` is the first member of `struct friend_adv`.

Reply:
That's true, but wouldn't you agree this is more "correct"? As this patch reintroduces the FRIEND_ADV, I don't think this change is out of scope.

Reply:
> but wouldn't you agree this is more "correct"

Not necessarily. The way the advertising buffer meta-data is designed is that all extensions of it have the generic part first. That's what `adv.c` internally also assumes (it's oblivious to any extensions). So if you'd risk breaking something with this then `adv.c` is also broken.

Reply:
I see, I'll revert.